numpy/core/tests/test_multiarray.py

```diff
@@ -5856,28 +5856,31 @@ def test_native_padding_2(self):
         self._check('^x3T{xi}', {'f0': (({'f0': ('i', 1)}, (3,)), 1)})
 
     def test_trailing_padding(self):
-        # Trailing padding should be included, *and*, the item size
-        # should match the alignment if in aligned mode
+        # Trailing padding should not be added according to struct spec.
         align = np.dtype('i').alignment
         size = np.dtype('i').itemsize
 
-        def aligned(n):
-            return align*(1 + (n-1)//align)
-
         base = dict(formats=['i'], names=['f0'])
 
-        self._check('ix',    dict(itemsize=aligned(size + 1), **base))
-        self._check('ixx',   dict(itemsize=aligned(size + 2), **base))
-        self._check('ixxx',  dict(itemsize=aligned(size + 3), **base))
-        self._check('ixxxx', dict(itemsize=aligned(size + 4), **base))
-        self._check('i7x',   dict(itemsize=aligned(size + 7), **base))
```

Comment:
I think these tests might remain valid with `T{...}`? In particular, I would expect trailing padding to be kept in a struct context, to match the behaviour of `sizeof(T)` in C, and what happens when structs are repeated

Reply:
Whereas the `ctypes` modules has set the precedent of ignoring all the remarks that the struct docs make about alignment...